addons/payment_adyen/models/payment_transaction.py

```diff
@@ -194,3 +195,53 @@ def _adyen_tokenize_from_feedback_data(self, data):
         _logger.info(
             "created token with id %s for partner with id %s", token.id, self.partner_id.id
         )
+
+    def _send_refund_request(self, **kwargs):
+        """ Override of payment to send a refund request to Adyen.
+        Note: self.ensure_one()
+        :return: None
+        """
+        super()._send_refund_request(**kwargs)
+        if self.provider != 'adyen':
+            return
+
+        refund_amount = kwargs.get('refund_amount', 0)
+
+        converted_amount = payment_utils.to_minor_currency_units(refund_amount,
+            self.currency_id, CURRENCY_DECIMALS.get(self.currency_id.name))
+
+        data = {
+            'merchantAccount': self.acquirer_id.adyen_merchant_account,
+            'modificationAmount': {
+                'value': converted_amount,
+                'currency': self.currency_id.name,
+            },
+            'originalReference': self.acquirer_reference,
+            'reference': self.reference,
+        }
+        response_content = self.acquirer_id._adyen_make_request(
+            url_field_name='adyen_payment_api_url',
+            endpoint='/refund',
+            payload=data,
+            method='POST'
+        )
+
+        # Handle the payment request response
+        if response_content['response'] == '[refund-received]':
+            # the PSP reference associated with this /refund request is different from the
+            # psp reference associated with the original payment request
+            tx = self._create_refund_transaction(
+                amount=-refund_amount,
+                original_transaction_id=self.id,
+                reference=self._compute_reference(self.provider, prefix=f"R{self.reference:s}"),
+            )
```

Comment:
The transaction **must** be created as soon as a payment flow is initiated, as it allows tracking the state of the payment/refund and storing meaningful information in case of failure.